### ISSUE TITLE
Move osoc blog somewhere simple

### ISSUE THREAD

**Issue by mietcls** [oSoc18/website#30]
The current blog is hosted on WordPress. People have referred to it as being a bit unsafe, and not easy to maintain. Suggestions were to move it to a regular markdown site or something else that's easy to maintain with git.

This would mean setting it up and making sure all the links remain the same.

**Comment by mietcls** [comment]
@jbelien you mentioned you have done this before? What would be your suggestions?
@jeborsel what were your arguments?
I'm also going to ping Astrid for her communication expertise once she has accepted to join the osoc repo.

**Comment by mietcls** [comment]
I like the idea of simple markdown site;
- gives the communications and other students the opportunity to learn about git
- lives in the "code" atmosphere
- integrates easier with the current site

On one condition; no compiling stuff or complicated things to run. Just plain html.. If possible.

**Comment by jbelien** [comment]
Yes, I already switched several website from WordPress to Jekyll (https://jekyllrb.com/).

And I really think it's the best option today :
- Nothing to setup
- Hostable on GitHub
- Students can easily post an new article through a Pull Request
- No need to create new WordPress account ; the students already have a GitHub account.

Design style, it's rather "easy" : 
- Create a template for the article list ;
- Create a template for a post ;

And then we have to migrate all the existing post to Markdown ; it's rather easy but it's work.
We can share that work though and I definitely will help to migrate the articles :)

**Comment by jbelien** [comment]
I forgot to mention that the blog can (should ?) be included inside oSoc website.
Since we fork the previous website anyway, the old articles will just follow next year :)

**Comment by mietcls** [comment]
@jbelien I've worked with jekyll in the past and I remember having to run things — but maybe it's changed?

I think it should be included in the Osoc website (any other opinions?) Would you like to set it up @jbelien ? I've had a huge fight with jekyll before and we aren't friends again yet..

**Comment by jbelien** [comment]
I don't mind helping setting it up of course :)

And sorry, I thought the new oSoc website is using Jekyll ... but apparently not 😝 
If we use GitHub pages to host the "blog" (and why not the website) we don't need to set up anything ; just enjoy the GitHub magic 😄 

And since the website don't current use Jekyll but because Jekyll can also serve  HTML static pages, I think we can use the actual structure as is (with maybe some small tweaks) and *just add* the "blog".

I'll put that on my todo list ASAP.
I'll fork this repo and try to host everything (including some blogposts) on GitHub pages and then we will be able to assess if it worth it or not.

**Comment by jbelien** [comment]
And voilà : https://jbelien.github.io/website-oSoc/blog.html

I obviously didn't focus on the layout/design (wouldn't make a lot of sense 😝).
I copied 2 articles (only the text so far) but it would be the same for all other articles of course. It's just to show you how the structure works.

If you have a look at https://github.com/jbelien/website-oSoc, you'll see :
- a [`_posts` folder](https://github.com/jbelien/website-oSoc/tree/master/_posts) : that's where all the blog posts will go (Markdown format) ;
- other Jekyll folders (like `_includes` and `_layouts`) ; more information here : https://jekyllrb.com/docs/structure/ ;

**EDIT :**
Even though GitHub Pages/Jekyll doesn't have any it to serve HTML pages. If we decide to switch to that solution, I would like to "clean" the HTML code and use Jekyll power such as [includes](https://jekyllrb.com/docs/includes/) for instance. I already prepared and used the [`_layouts\defaut.html`](https://github.com/jbelien/website-oSoc/blob/master/_layouts/default.html) for the [`blog.html`](https://github.com/jbelien/website-oSoc/blob/master/blog.html) page.

What do you think about it ?
Do you have any question ?

---

PS : I also discovered that there is WordPress migration plugin for Jekyll : https://import.jekyllrb.com/docs/wordpress/. I never used it but that should ease the migration.

**Comment by mietcls** [comment]
That looks great! How would you keep the URL's the same btw? So we don't lose links? Let's talk about it on Thursday.

**Comment by jbelien** [comment]
> How would you keep the URL's the same btw? So we don't lose links?

Not an issue at all.
As you can see 
http://open.summerofcode.be/2018/07/25/use-bike-brussels-route-planner-instead-google-maps/
is now
https://jbelien.github.io/website-oSoc/2018/07/25/use-bike-brussels-route-planner-instead-google-maps.html
but we can remove the `.html` at the end by setting [`_config.yml`](https://github.com/jbelien/website-oSoc/blob/master/_config.yml#L2) up.

**Comment by jbelien** [comment]
@MichielLeyman :
> Ismaila is interning at OKBE and just finished making the current version in React, with the idea of now building functionality on top (forms on the website itself, student “portal”, …)

So I deleted my `jbelien/website-oSoc` repo.
